Slime plushes, rainbow flush plush and related changes

Gilgaxx opened the pull request:
## About The Pull Request
This PR adds a total of 6 new plushes (slimeplush_pink, slimeplush_green, slimeplush_blue, slimeplush_red, slimeplush_rainbow and flushplush_rainbow)
![fCaWh3rS82](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/6ee4b0f0-354f-4452-8d2c-3404b3c6ee6b)
Added said plushes to relevant places:
* Plush order beacon
* Arcade and Warping gold rune (added the flushplush_rainbow and the new slimeplushie/random to both of these)
* Cargo plush crate (added new plushs to the normal plush crate, removed the dehydrated space carp form the normal crate, added a new contraband only crate without moths but with the dehydrated carp (Plushie Crate Without Moth Plushies))
* Mail (RD and scientists can get slime plushes, curators can get the flushplush_rainbow rarely)
* Family heirloom (added the new slime variants for RD and scientist)
* Phobia (Alien phobia has the slime mob as part of the phobia but not the slime plush, so that was added (to match the lizard phobia working with the lizard plushie))

Also made some minor grammar fixes to the moth plush descriptions.

## Why It's Good For The Game

Before this PR, getting a random plush usually meant getting a large number of moth plushes (25 out of 43 plushes). So by adding more slime plushes and a new flush plush, we lower the odds of getting the moth plushes. 
The addition of the crate is seen as a fix to the original containing contraband, adding a new contraband crate without the moth plushes.
Along the way, put the new slime plushes in someplaces, put slime plushes in places that they weren't before (like the arcade and phobia), also fixing grammar is always nice :)

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

* Warping gold rune
![dreamseeker_YFkmxxkSNZ](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/3b857963-370b-4d81-b6f8-eb54afeae917)

* Cargo plush crate - Before emagging the console
![dreamseeker_8R0lqgLLPn](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/5c7d9e98-3892-44e2-bb50-f916d7e4e02b)
After emagging the console
![dreamseeker_ADBHZyIAmC](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/617d39f2-85e6-473a-9876-755baa406a9f)
Opened crate
![dreamseeker_wUq9qztSA1](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/20900d97-c61b-48c3-a13f-129b2629ef96)

* Mail - RD opening the mail (and getting a green slime)
![dreamseeker_8GucNSKleE](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/f6b7ee27-a63e-41e1-9ef0-6108ce66eb32)

* Family heirloom - Spawning as a scientist (also getting a green slime)
![dreamseeker_pjNJSqHGAR](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/31380832-41fb-4538-886a-ffc680a0a70d)

* Phobia - You can see in chat that the slime plush is making the person unconfortable
![dreamseeker_bisB6ixtMt](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/3404fb91-82e0-4baf-aa77-8eb6d38eae54)

</details>

## Changelog
:cl:
add: Added 5 new slime plushies
add: Added a rainbow flush plush
tweak: Removed dehydrated carps from the plushes crate
add: Added a new contraband crate: "Plushie Crate Without Moth Plushies". Dehydrated carps included, moth plushes excluded.
tweak: Alien phobia now includes slime plushes (it already included slime mobs).
spellcheck: There's no such thing as plushys, they're plushies now.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Gilgaxx committed:
Adds Slime Plushes, a rainbow flushed plush and others

Add Plushes (with sprites). Add plushes to arcade, warping gold rune, family heirloom, plush beacon, phobia and mail. Add new plush crate without moths (contraband). Fixes some grammar in the moth plushes.

Gilgaxx committed:
Suicide verb and fix for slime plush

The suicide verb for slime plushes was added to match the moth plush. Also fixed the problem with the family heirloom spawning.

Gilgaxx committed:
minor spelling mistakes

An extra space removed and the icon state on the improper random slimeplush was fixed (just in case)

EvilDragonfiend commented on code/modules/cargo/packs.dm:
```diff
@@ -3108,9 +3108,26 @@
+	var/plush_nomoth
+	for(var/i in 1 to num_contained)
+		plush_nomoth = pick(subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth))
+		new plush_nomoth(C)
```
cache this subtype list into a datum variable. You're recalculating a list every this loop.
or at least you should declare a temporary variable right above of line 3127

Gilgaxx replied:
I'm not sure what "cache this subtype list into a datum variable" means, I'm still a bit new to this, could you give an example or explain with layman terms?
Also, line 3126 is `var/plush_nomoth`. Is that not a declaration of a temporary variable?

EvilDragonfiend replied:
I made another comment to explain this.

EvilDragonfiend commented on code/modules/cargo/packs.dm:
```diff
@@ -3108,9 +3108,26 @@
 	var/plush
 	for(var/i in 1 to num_contained)
-		plush = pick(subtypesof(/obj/item/toy/plush))
+		plush = pick(subtypesof(/obj/item/toy/plush) - /obj/item/toy/plush/carpplushie/dehy_carp)
```
this too

EvilDragonfiend commented on code/modules/cargo/packs.dm:
```diff
@@ -3108,9 +3108,26 @@
+	var/plush_nomoth
+	for(var/i in 1 to num_contained)
+		plush_nomoth = pick(subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth))
+		new plush_nomoth(C)
```
These lines actually mean:
```dm
	var/plush_nomoth
	for(var/i in 1 to num_contained)
		var/_temporary_list = subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth)
		plush_nomoth = pick(_temporary_list )
		new plush_nomoth(C)
```
You can see `var/_temporary_list = subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth)` is in the loop. This means you're creating a list of items every time this loop is continued.
but calculation can be done for the first time.
```dm
	var/plush_nomoth
	var/_temporary_list = subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/moth)
	for(var/i in 1 to num_contained)
		plush_nomoth = pick(_temporary_list )
		new plush_nomoth(C)
```
Now you don't calculate the list again and again.

Gilgaxx replied:
Oh, that makes sense! I'll change both of the crates so they work like that

Gilgaxx replied:
Alright a new commit should be up.

<details>
<summary>And here is me spawning one of each crate with the cargo spawn verb:</summary>

![dreamseeker_N2psD7B3rX](https://github.com/BeeStation/BeeStation-Hornet/assets/105242196/d02196ac-31ad-4c5b-a590-7c02e593bd00)
</details>
The runtime happened a few seconds after loading and I don't believe it's related to this PR, do correct me if I'm wrong!

EvilDragonfiend replied:
yeah it's unrelated. no worries.

Gilgaxx committed:
Declared temporary variable for the crates

Declared temporary variable for both crates added as per EvilDragonfiend's request, so it doesn't calculate the list every loop.

EvilDragonfiend reviewed (approved):
Pigeon requested me checking if we can merge suicide procs (since slime and moth ones are the same), but I do not see it works here.
I will just approve it and let other maints look on it in this case.
